docs(messaging, android): update requesting permissions documentation (#11554)
firebase/flutterfire: docs/cloud-messaging/receive.md

- On Android, if the user force-quits the app from device settings, it must be manually reopened for messages to start working.
- On web, you must have requested a token (using `getToken()`) with your web push certificate.

## Request permission to receive messages

> On iOS, macOS, web and android (android =>13. Lower versions do not require permission), before FCM payloads can be received on your device,

Reviewer: ```suggestion
On iOS, macOS, web and Android 13 (or newer), before FCM payloads can be received on your device,
```

- `notDetermined`: The user has not yet chosen whether to grant permission.
- `provisional`: The user granted provisional permission

> Note: On Android <=12 `authorizationStatus` will return `authorized` if the user has not disabled notifications for the app via the operating systems settings.

Reviewer: ```suggestion
Note: On Android versions prior to 13, `authorizationStatus` will return `authorized` if the user has not disabled notifications for the app via the operating system settings.
```
